### PR TITLE
Better handling of Inline schema

### PR DESCRIPTION
**Pull request by wing328**
Better handling of Inline schema based on feedback from the users.

allOf inline schemas (children) are not created by default.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by devhl-labs** [comment]
This PR does not resolve the following:
```yaml
openapi: 3.0.1
components:
  schemas:
    Language:
      type: object
      properties:
        name:
          type: string
          readOnly: true
    Clan:
      type: object
      properties:
        chatLanguage:
          # nullable: true
          allOf:
          - $ref: '#/components/schemas/Language'
```

image
Note the two different language classes. This was not an issue in 6.0.0 beta 914b539eea7097ad2b4323d20c8f95144782cb68

File: samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/src/serializers.dart [hunk 1]
```diff
@@ -75,12 +73,10 @@ part 'serializers.g.dart';
   ArrayTest,
   Capitalization,
   Cat,
-  CatAllOf,$CatAllOf,
```
Comment (ahmednfwela) [inline samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/src/serializers.dart:78]: looking at the dart samples in 
https://github.com/OpenAPITools/openapi-generator/tree/inline-schema-improve3/samples/openapi3/client/petstore/dart-dio

I see that [the inline model files still remain](https://github.com/OpenAPITools/openapi-generator/blob/inline-schema-improve3/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/src/model/cat_all_of.dart) but  their usage is removed

what might be causing this ?

Comment (wing328) [reply samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/src/serializers.dart:78]: https://github.com/OpenAPITools/openapi-generator/blob/inline-schema-improve3/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/src/model/cat_all_of.dart is left over from previous code generation.

I can confirm manually removing https://github.com/OpenAPITools/openapi-generator/blob/inline-schema-improve3/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/src/model/cat_all_of.dart and regenerating dart samples will recreate this file, which is the expected behaviour as part of this change.

**Comment by wing328** [comment]
UPDATE: started a new PR (https://github.com/OpenAPITools/openapi-generator/pull/15682) using a different approach

**Comment by wing328** [comment]
UPDAT: https://github.com/OpenAPITools/openapi-generator/pull/15682 has been merged.

**Pull request closed without merging**